numba/array_analysis.py

```diff
@@ -1422,7 +1432,59 @@ def _analyze_op_call_numpy_reshape(self, scope, equiv_set, args, kws):
             typ = self.typemap[args[1].name]
             if isinstance(typ, types.BaseTuple):
                 return args[1], []
-        return tuple(args[1:]), []
+
+        # Reshape is allowed to take one argument that has the value <0.
+        # This means that the size of that dimension should be inferred from
+        # the size of the array being reshaped and the other dimensions
+        # specified.  Our general approach here is to see if the reshape
+        # has any <0 arguments.  If it has more than one then throw a
+        # ValueError.  If exactly one <0 argument is found, remember its
+        # argument index.
+        stmts = []
+        neg_one_index = -1
```

Comment:
this could probably be `neg_index` as its not just `-1` that triggers this behaviour.

Reply:
This is keeping track of the index of a negative value if one is found.  It isn't the value that we are looking for.  I already changed it to look for any negative value.